File: app/src/androidTest/java/io/github/fvasco/pinpoi/importer/GpxImporterTest.java
package io.github.fvasco.pinpoi.importer;

import org.junit.Test;

import java.util.List;

import io.github.fvasco.pinpoi.model.Placemark;

/**
 * @author Francesco Vasco
 */
public class GpxImporterTest extends AbstractImporterTestCase {

    @Test
    public void testImportImpl() throws Exception {
        final List<Placemark> list = importPlacemark(new GpxImporter(), "test.gpx");

        assertEquals(2, list.size());

        final Placemark p = list.get(0);
        assertEquals("Test", p.getName());
        assertEquals("descTest", p.getDescription());
        assertEquals(1, p.getLatitude(), 0.1);
        assertEquals(2, p.getLongitude(), 0.1);
    }
}

File: app/src/androidTest/java/io/github/fvasco/pinpoi/importer/GpxImporterTest.kt
package io.github.fvasco.pinpoi.importer

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith

/**
 * @author Francesco Vasco
 */
@RunWith(AndroidJUnit4::class)
class GpxImporterTest : AbstractImporterTestCase() {

    @Test
    fun testImportImpl() {
        val list = importPlacemark(GpxImporter(), "test.gpx")

        assertEquals(2, list.size)

        val p = list[0]
        assertEquals("Test", p.name)
        assertEquals("descTest", p.description)
        assertEquals(1.0, p.coordinates.latitude.toDouble(), 0.1)
        assertEquals(2.0, p.coordinates.longitude.toDouble(), 0.1)
    }
}